.github/workflows/build.yml

```diff
@@ -25,10 +25,23 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
     - name: 'Install Dependencies'
-      run: sudo apt-get update && make ciprepare
+      run: |
+        sudo apt-get update && make ciprepare
+        sudo apt-get install libssl-dev
+        cargo install cargo-tarpaulin
     - name: 'Run rust tests'
       run: make --keep-going test
+    - name: 'Generate coverall coverage'
+      run: make --keep-going coverage
+    - name: 'rename coverall file'
+      run: make movefile
+    - name: 'upload the xml coverage to coverall server'
```

Comment:
either remove "coverall" or change it into "codecov" - they are two different services